src/query/storages/fuse/src/operations/agg_index_sink.rs

```diff
@@ -68,12 +77,36 @@ impl AggIndexSink {
             keep_block_name_col,
             location_data: HashMap::new(),
             blocks: vec![],
+            max_memory_usage,
         });
 
         Ok(ProcessorPtr::create(sinker))
     }
 
-    fn process_block(&mut self, block: &mut DataBlock) {
+    fn check_memory_limit(&self, block: &DataBlock) -> Result<()> {
```

Comment:
We have the soft oom by default, does it also working?

Reply:
Yup it works, I have add tests here: https://github.com/datafuselabs/databend/pull/14140/files#diff-4001267ac227fed7b868cb28249ed3020c48bf6c843cb120e213ac816cdbd092

Reply:
So we don't need `check_memory_limit`?

Reply:
If we don't check memory here, how do we return the corresponding error message?

Reply:
No need return, we have many places to return soft oom.

Reply:
This is the soft oom error:
<img width="660" alt="image" src="https://github.com/datafuselabs/databend/assets/172204/d17be3f7-b97c-4dca-84d8-a44944ff941d">